docker_fargate/docker_fargate_stack.py

```diff
@@ -93,28 +93,20 @@ def __init__(self, scope: Construct, context: str, env: dict, vpc: ec2.Vpc, **kw
             ssl_policy=elbv2.SslPolicy.FORWARD_SECRECY_TLS12_RES, # Strong forward secrecy ciphers and TLS1.2 only.
         )
 
-        # Overriding health check timeout helps with sluggishly responding app's (e.g. Shiny)
-        # https://docs.aws.amazon.com/cdk/api/v1/python/aws_cdk.aws_elasticloadbalancingv2/ApplicationTargetGroup.html#aws_cdk.aws_elasticloadbalancingv2.ApplicationTargetGroup
-        load_balanced_fargate_service.target_group.configure_health_check(interval=Duration.seconds(120), timeout=Duration.seconds(60))
-
-        if False: # enable/disable autoscaling
-            scalable_target = load_balanced_fargate_service.service.auto_scale_task_count(
-               min_capacity=1, # Minimum capacity to scale to. Default: 1
-               max_capacity=4 # Maximum capacity to scale to.
-            )
-
-            # Add more capacity when CPU utilization reaches 50%
-            scalable_target.scale_on_cpu_utilization("CpuScaling",
-                target_utilization_percent=50
-            )
-
-            # Add more capacity when memory utilization reaches 50%
-            scalable_target.scale_on_memory_utilization("MemoryScaling",
-                target_utilization_percent=50
-            )
-
-            # Other metrics to drive scaling are discussed here:
-            # https://docs.aws.amazon.com/cdk/api/v1/python/aws_cdk.aws_autoscaling/README.html
+        scalable_target = load_balanced_fargate_service.service.auto_scale_task_count(
+           min_capacity=1, # Minimum capacity to scale to. Default: 1
+           max_capacity=4 # Maximum capacity to scale to.
+        )
+
+        # Add more capacity when CPU utilization reaches 50%
+        scalable_target.scale_on_cpu_utilization("CpuScaling",
+            target_utilization_percent=50
+        )
+
+        # Add more capacity when memory utilization reaches 50%
+        scalable_target.scale_on_memory_utilization("MemoryScaling",
+            target_utilization_percent=50
+        )
```

Comment:
are you sure this works?  Fargate requires a match set of CPU and memory, I'm not sure they can be scaled separately.  https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task-cpu-memory-error.html

Reply:
> Fargate requires a match set of CPU and memory

Yes, that's true, for a *task definition*.  The CPU and memory defined for the core task is given at lines 83 and 87, above, and they are listed as valid combinations on the page you linked.

*This* part of the script describes the *scaling*, i.e. at what thresholds of CPU or memory usage ECS should add more copies of the task (i.e., should scale horizontally).

> are you sure this works? 

We will have to monitor the load on the system, see how it scales and possibly do some tuning.  We do this as part of our weekly Synapse stack review meeting (though we will have to add the ECS metrics for this new deployment to our dashboard).